### PR TITLE
205

### LINKED ISSUE #205: "new project" button

### PR DESCRIPTION
**Pull request by thedavidmeister**
fixes #205

### CODE REVIEW

File: sass/mappings/blocks/_list.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-main > div {
+[list-id] {
```
Comment (thedavidmeister) [inline sass/mappings/blocks/_list.scss]: this DOM structure changed a bit

File: src/cljs/app/datascript/state.cljs [hunk 2]
```diff
@@ -8,7 +8,14 @@
...
 
-(def conn (local-storage (du/conn-cell) ::conn-state))
+(def schema {})
+
+(def conn (local-storage (du/conn-cell schema) ::conn-state))
+
+(defn new-conn!
+  "Sets the state conn back to an empty db"
+  []
+  (reset! conn (d/empty-db schema)))
```
Comment (thedavidmeister) [inline src/cljs/app/datascript/state.cljs]: pretty heavy handed... but it does the job :)

File: src/cljs/app/pages/index.cljs [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
+    [app.elements.new-project :as new-project]))
 
-(c/defelem page
+(h/defelem page
   [attributes children]
-  (map #(lists/item-list :list-id % attributes children) (range 1 6)))
+  (h/div
+    :id "index"
+
+    (new-project/new-project-button)
```
Comment (thedavidmeister) [inline src/cljs/app/pages/index.cljs]: this seems as good a place as any to put the new project button...

File: test/hl/app/elements/auth0_test.clj [hunk 4]
```diff
@@ -1,4 +1,4 @@
-(ns app.auth0-test
+(ns app.elements.auth0-test
```
Comment (thedavidmeister) [inline test/hl/app/elements/auth0_test.clj]: I just moved this file to make it more explicit that we're testing an element here

File: test/hl/app/elements/new_project_test.clj [hunk 5]
```diff
@@ -0,0 +1,28 @@
+(ns app.elements.new-project-test
```
Comment (thedavidmeister) [inline test/hl/app/elements/new_project_test.clj]: new tests for the new button

File: test/hl/app/index_test.clj [hunk 6]
```diff
@@ -3,62 +3,16 @@
...
     [clojure.test.check.generators :as gen]))
 
 (use-fixtures :each fixtures/selenium-driver!)
 
-(def page-selector "[list-id]")
+(def page-selector "#index")
 (def page-hash "#")
 (def page-title "Estimate Work")
 
-(def inputs-selector (str page-selector " input"))
```
Comment (thedavidmeister) [inline test/hl/app/index_test.clj]: I moved all this to a new namespace so I could re-use it easily in the new tests

File: test/hl/app/index_test.clj [hunk 7]
```diff
@@ -3,62 +3,16 @@
...
     [clj-webdriver.taxi])
   (:require
+    [util.test-inputs :as tinp]
     [util.test-fixtures :as fixtures]
     [clojure.test.check.generators :as gen]))
 
 (use-fixtures :each fixtures/selenium-driver!)
 
-(def page-selector "[list-id]")
+(def page-selector "#index")
```
Comment (thedavidmeister) [inline test/hl/app/index_test.clj]: this DOM changed slightly

File: test/hl/app/index_test.clj [hunk 8]
```diff
@@ -71,56 +25,56 @@
 
 (deftest first-input-delete
   "Test that the first input can be deleted, which selects the new first input"
-  (make-some-inputs)
+  (tinp/make-some-inputs)
```
Comment (thedavidmeister) [inline test/hl/app/index_test.clj]: all this `tinp` stuff is just namespacing...

File: test/hl/util/test_inputs.clj [hunk 9]
```diff
@@ -0,0 +1,53 @@
+(ns util.test-inputs
```
Comment (thedavidmeister) [inline test/hl/util/test_inputs.clj]: this is the new namespace for input wrangling functions

**Comment by meelijane** [comment]
i think this button needs to be in a div where we can put other controls..

**Pull request merged**